kratos/python/add_geometrical_utilities_to_python.cpp

```diff
@@ -192,19 +191,31 @@ void AddGeometricalUtilitiesToPython(pybind11::module &m)
         .def("CreateNodesArray", &IsosurfacePrinterApplication::CreateNodesArray)
         ;
 
-    //binbased locators
+    // BinBasedFastPointLocator
     py::class_< BinBasedFastPointLocator < 2 >, BinBasedFastPointLocator < 2 >::Pointer >(m,"BinBasedFastPointLocator2D")
         .def(py::init<ModelPart& >())
         .def("UpdateSearchDatabase", &BinBasedFastPointLocator < 2 > ::UpdateSearchDatabase)
         .def("UpdateSearchDatabaseAssignedSize", &BinBasedFastPointLocator < 2 > ::UpdateSearchDatabaseAssignedSize)
         .def("FindPointOnMesh", &BinBasedFastPointLocator < 2 > ::FindPointOnMeshSimplified)
+        .def("FindPointOnMesh", [](BinBasedFastPointLocator < 2 >& rSelf, const array_1d<double,3>& rCoords ){
+            Element::Pointer p_elem = nullptr;
+            Vector N;
+            const bool is_found = rSelf.FindPointOnMeshSimplified(rCoords,N,p_elem);
+            return std::tuple<bool,Vector,Element::Pointer>{is_found,N,p_elem};
+        })
         ;
 
     py::class_< BinBasedFastPointLocator < 3 >, BinBasedFastPointLocator < 3 >::Pointer >(m,"BinBasedFastPointLocator3D")
         .def(py::init<ModelPart&  >())
         .def("UpdateSearchDatabase", &BinBasedFastPointLocator < 3 > ::UpdateSearchDatabase)
-        .def("FindPointOnMesh", &BinBasedFastPointLocator < 3 > ::FindPointOnMeshSimplified)
         .def("UpdateSearchDatabaseAssignedSize", &BinBasedFastPointLocator < 3 > ::UpdateSearchDatabaseAssignedSize)
+        .def("FindPointOnMesh", &BinBasedFastPointLocator < 3 > ::FindPointOnMeshSimplified)
```

Comment:
What's the difference with this one?

Reply:
The dimension

Reply:
no, not just the dimension. To use this one one would need to pass a pointer to the element to be returned. To be honest i don't have a clue about how we can accomplish this from python. That is, from what i see that function cannot be used.
I am not removing it since someone might be using it...(although i don't know how, otherwise i would have done the same)

in any case the new interface is way simpler:

~~~py
[found,N,pelem] = locator.FindPointOnMesh(pos)
~~~

(where N and pelem only make sense if found==True, the same way as would happen in c++)